autoreject/utils.py

```diff
@@ -43,21 +43,22 @@ def _check_data(epochs, picks, ch_constraint='data_channels',
         mne.io.meas_info.channel_type(picked_info, idx)
         for idx in range(len(picks))}
 
-    if not _check_ch_locs(picked_info['chs']):
+    if check_loc and not _check_ch_locs(picked_info['chs']):
         raise RuntimeError('Valid channel positions are needed '
                            'for autoreject to work')
 
     # XXX : ch_constraint -> allow_many_types=True | False
     if ch_constraint == 'data_channels':
-        if not all(ch in ('mag', 'grad', 'eeg', 'hbo', 'hbr')
+        if not all(ch in ('mag', 'grad', 'eeg', 'hbo', 'hbr', 'ecog', 'seeg')
                    for ch in ch_types_picked):
-            raise ValueError('AutoReject only supports mag, grad, and eeg '
-                             'at this point.')
+            raise ValueError('AutoReject only supports mag, grad, eeg, and '
+                             'ecog at this point.')
     elif ch_constraint == 'single_channel_type':
         if sum(ch in ch_types_picked for ch in ('mag', 'grad', 'eeg',
-                                                'hbo', 'hbr')) > 1:
-            raise ValueError('AutoReject only supports mag, grad, and eeg '
-                             'at this point.')  # XXX: to check
+                                                'hbo', 'hbr', 'ecog', 'seeg'))\
```

Comment:
backslashes are really ugly ... can you avoid it with proper use of paranthesis?

Reply:
updated